src/main/java/edu/umich/tl/ApiCallHandler.java

```diff
@@ -8,46 +8,115 @@
 import org.apache.http.HttpResponse;
 import org.apache.http.NameValuePair;
 import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpDelete;
 import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
 import org.apache.http.client.methods.HttpUriRequest;
 import org.apache.http.impl.client.DefaultHttpClient;
 import org.apache.http.message.BasicNameValuePair;
 
+import com.google.common.base.Stopwatch;
+
 import edu.umich.tl.CourseDelete.CanvasCallEnum;
 
 public class ApiCallHandler {
+	private static final String DELETE = "DELETE";
+
+	private static final String GET = "GET";
+
 	private static Log M_log = LogFactory.getLog(ApiCallHandler.class);
+	
 	private String canvasURL=CourseDelete.canvasURL;
 	private String esbURL=CourseDelete.esbURL;
 	private CanvasCallEnum canvasCall;
 	private static String canvasToken=CourseDelete.canvasToken;
 	private static final String PER_PAGE = "per_page=100";
 	private static final String API_VERSION = "/api/v1";
+	private static final String COURSES = "/courses/";
+	private static final String ACCOUNTS = "/accounts/1";
 	
 	public ApiCallHandler(CanvasCallEnum canvasCall) {
 		this.canvasCall=canvasCall;
 	}
 	
 	public enum RequestTypeEnum{
-		TERM, UNPUBLISHED_COURSE_LIST, UNPUBLISHED_COURSE_LIST_PAGINATION_URL;
+		TERM, UNPUBLISHED_COURSE_LIST, UNPUBLISHED_COURSE_LIST_PAGINATION_URL, 
+		ASSIGNMENT,ANNOUNCEMENT,CONFERENCE,DISCUSSION_TOPICS,FILES,GRADE_CHANGES,GROUPS,MODULES,
+		PAGES,QUIZZES,EXTERNAL_TOOLS,COURSE_AUDIT_LOGS,COURSE_AUDIT_LOGS_PAGINALTION_URL,COURSE_DELETE;
 	}
 
-	public HttpResponse getApiResponse(RequestTypeEnum requestType, String canvasTermIdForSisTermId, String url) {
+	public HttpResponse getApiResponse(RequestTypeEnum requestType, String canvasTermIdForSisTermId, String url,String courseId) {
 		HttpResponse httpResponse = null;
 		String urlSuffix;
 		switch (requestType) {
 		case TERM:
```

Comment:
Consider using the case statement to just set the urlSuffix, boolean, and request type and then just having one call to the urlConstructorAndCanvasCallManager.  That would let you default the boolean and request type to the common values and, IMHO, make it easier to tell what different about each call.  I.e. the call with boolean false and / or method not equal to GET would be much more obvious.